### DISCUSSION TITLE
dev, staging and prod on now

### DISCUSSION THREAD

**Discussion by Manubi** [vercel/next.js#10841] [Help]
Hey, 
what is the best solution to get a setup where I can set different env variables for the different environments and can deploy them to different urls such as http://dev.my-domain.com, http://staging.my-domain.com...

Ideally if i push the dev branch into the staging branch it will push it to http://staging.my-domain.com

How do you guys do it?

**Comment by paulogdm** [comment]
Hello @Manubi ,

The team is currently working on this and it should arrive in the upcoming weeks. The idea is to offer "multi-env" support as you described. Perhaps [this](https://zeit.co/docs/v2/custom-domains#assigning-a-domain-to-a-git-branch) can assist you while we don't release this.

**Reply by Jakkc** [reply]
It would be really appreciated if the docs could have been updated to reflect this. Just a little pointer along the lines of 'this feature is completely standard but we don't support it yet - it's coming soon though so don't worry!'. I spent the past 24 hours trying crack pot solutions online and I'd like that time back!

**Reply by paulogdm** [reply]
You can always send us an email (support@zeit.co) and we will certainly give you a hand.

**Reply by Vadorequest** [reply]
@Jakkc Take a look at https://github.com/UnlyEd/next-right-now, it's built with such workflow in mind. It may be "too much" but you can use it as a learning resource still :)

(and the Zeit team is really great at helping out too!)

**Comment by Vadorequest** [comment]
@Manubi Check out https://github.com/UnlyEd/next-right-now, it does exactly that.

Maybe you can use it as a boilerplate if you're starting from scratch, or learning from how we did things to support multi-env. :)
